internal/lsp/server.go

```diff
@@ -680,17 +680,26 @@ func (l *LanguageServer) handleTextDocumentCompletion(
 	_ context.Context,
 	_ *jsonrpc2.Conn,
 	req *jsonrpc2.Request,
-) (result any, err error) {
+) (any, error) {
 	var params types.CompletionParams
 	if err := json.Unmarshal(*req.Params, &params); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal params: %w", err)
 	}
 
+	// items is allocated here so that the return value is always a non-nil CompletionList
 	items, err := l.completionsManager.Run(params, &providers.Options{RootURI: l.clientRootURI})
 	if err != nil {
 		return nil, fmt.Errorf("failed to find completions: %w", err)
 	}
 
+	if len(items) == 0 {
```

Comment:
not strictly related to this PR, but this makes sure that the items is never null.